### PR TITLE
Video topic question service

### PR DESCRIPTION
**Pull request by BalogDorin**
pull request

### CODE REVIEW

File: question-service/src/main/java/com/information/center/questionservice/controller/AnswerController.java [hunk 1]
```diff
@@ -8,40 +8,37 @@
...
 @Component
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class AnswerController implements AnswerEndpoint {
 
   private final AnswerService answerService;
 
-  @Override
-  public AnswerResponse create(@RequestBody AnswerRequest answerRequest,
-      @PathVariable("questionExternalId") String questionExternalId) {
+  @PostMapping("/question/{questionExternalId}")
```
Comment (kolyaattila) [inline question-service/src/main/java/com/information/center/questionservice/controller/AnswerController.java:19]: `@PutMapping` and others are already in the interface, is no necessary to put them here to

File: question-service/src/main/java/com/information/center/questionservice/controller/QuestionController.java [hunk 2]
```diff
@@ -1,61 +1,65 @@
...
 
-  @Override
-  public QuestionResponse create(@RequestBody QuestionRequest questionRequest,
-      @PathVariable("topicExternalId") String topicExternalId) {
+    private final QuestionValidateService questionValidateService;
 
-    return questionService.create(questionRequest, topicExternalId);
-  }
+    @PostMapping("/topic/{topicExternalId}")
+    public QuestionResponse create(@RequestBody QuestionRequest questionRequest, @PathVariable("topicExternalId") String topicExternalId) {
```
Comment (kolyaattila) [inline question-service/src/main/java/com/information/center/questionservice/controller/QuestionController.java:28]: Same here with the mapping anotation

File: question-service/src/main/java/com/information/center/questionservice/model/response/QuestionResponse.java [hunk 3]
```diff
@@ -23,4 +25,6 @@ public QuestionResponse() {
   private QuestionDifficulty questionDifficulty;
 
   private String topicExternalId;
+
+  private Date startDate;
```
Comment (kolyaattila) [inline question-service/src/main/java/com/information/center/questionservice/model/response/QuestionResponse.java:29]: why are you need start date here

File: question-service/src/main/java/com/information/center/questionservice/service/QuestionServiceImpl.java [hunk 4]
```diff
@@ -2,122 +2,139 @@
...
 
-    String questionExternalId = UUID.randomUUID().toString();
+    @Value("${endpoints.topic:http://localhost:8881}")
+    private String equipmentEndpoint;
 
-    for (AnswerRequest answerRequest : questionRequest.getAnswers()) {
-      answerRequest.setQuestionExternalId(questionExternalId);
-      answerRequest.setExternalId(UUID.randomUUID().toString());
+    @Override
+    public String getTopicNameByExternalId(String topicId) {
```
Comment (kolyaattila) [inline question-service/src/main/java/com/information/center/questionservice/service/QuestionServiceImpl.java:45]: if you want to call the topic-service you can use `@FeignClient`. You can find some example in account-service

File: question-service/src/main/resources/db/data/1.0/initial_schema.yaml [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
-      changes:
-        - sqlFile:
-            dbms: postgresql
-            encoding: utf8
-            endDelimiter: \nGO
-            path: difficulty_enum.sql
-            relativeToChangelogFile: true
-            splitStatements: true
-            stripComments: true
+#  - changeSet:
```
Comment (kolyaattila) [inline question-service/src/main/resources/db/data/1.0/initial_schema.yaml:2]: why did you com this

File: question-service/src/main/resources/db/data/1.0/question.sql [hunk 6]
```diff
@@ -4,5 +4,5 @@ create table public.Question (
     created           timestamp,
     external_id       varchar(255) not null unique,
     topic_id          varchar(255) not null,
-    difficulty        Difficulty not null
+    difficulty        varchar(255) not null
```
Comment (kolyaattila) [inline question-service/src/main/resources/db/data/1.0/question.sql:7]: why did you change this

File: topic-service/src/main/java/com/information/center/topicservice/model/request/TopicRequest.java [hunk 7]
```diff
@@ -9,6 +9,6 @@
 @NoArgsConstructor
 public class TopicRequest {
 
-  private String name;
+    private String name;
```
Comment (kolyaattila) [inline topic-service/src/main/java/com/information/center/topicservice/model/request/TopicRequest.java:12]: here you just format the code

File: video-service/src/main/java/com/information/center/videoservice/service/VideoServiceImpl.java [hunk 8]
```diff
@@ -6,57 +6,144 @@
...
-  public VideoEntity findById(String externalId) {
-
-    return videoRepository.findByUid(externalId)
-        .orElseThrow(throwNotFoundItem("video", externalId));
-  }
-
-  private Supplier<InconsistentDataException> throwNotFoundItem(String item, String itemId) {
-    return () -> new InconsistentDataException("Cannot find " + item + " by id " + itemId);
-  }
+    private final VideoRepository videoRepository;
```
Comment (kolyaattila) [inline video-service/src/main/java/com/information/center/videoservice/service/VideoServiceImpl.java:34]: - you have no tests
- make some refactoring

File: question-service/src/main/resources/db/data/1.0/initial_schema.yaml (hunk 5 above)
Comment (BalogDorin) [reply question-service/src/main/resources/db/data/1.0/initial_schema.yaml:2]: we discouse that we can't define types in Postgress, at least not this case

File: question-service/src/main/resources/db/data/1.0/question.sql (hunk 6 above)
Comment (BalogDorin) [reply question-service/src/main/resources/db/data/1.0/question.sql:7]: Can't have custom type

File: topic-service/src/main/java/com/information/center/topicservice/model/request/TopicRequest.java (hunk 7 above)
Comment (BalogDorin) [reply topic-service/src/main/java/com/information/center/topicservice/model/request/TopicRequest.java:12]: I'm using a tool for code format

File: question-service/src/main/resources/db/data/1.0/initial_schema.yaml (hunk 5 above)
Comment (kolyaattila) [reply question-service/src/main/resources/db/data/1.0/initial_schema.yaml:2]: yes, i solved the problem. make a rebase

File: topic-service/src/main/java/com/information/center/topicservice/model/request/TopicRequest.java (hunk 7 above)
Comment (kolyaattila) [reply topic-service/src/main/java/com/information/center/topicservice/model/request/TopicRequest.java:12]: if you don't change anything in the class is no need to reformat

File: question-service/src/main/java/com/information/center/questionservice/client/TopicServiceClient.java [hunk 9]
```diff
@@ -0,0 +1,17 @@
+package com.information.center.questionservice.client;
+
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+@FeignClient(name = "topic-service")
```
Comment (kolyaattila) [inline question-service/src/main/java/com/information/center/questionservice/client/TopicServiceClient.java:8]: you have to add a fallBack method in case the service is not UP

**Comment by kolyaattila** [comment]
please make squash before you create the pull request

File: question-service/src/main/java/com/information/center/questionservice/controller/AnswerController.java [hunk 10]
```diff
@@ -8,40 +8,43 @@
...
 public class AnswerController implements AnswerEndpoint {
 
-  private final AnswerService answerService;
+    private final AnswerService answerService;
 
-  @Override
-  public AnswerResponse create(@RequestBody AnswerRequest answerRequest,
-      @PathVariable("questionExternalId") String questionExternalId) {
+    @Override
+    @PostMapping("/question/{questionExternalId}")
```
Comment (kolyaattila) [inline question-service/src/main/java/com/information/center/questionservice/controller/AnswerController.java:20]: It's not necessary to add `Mapping` anotations in here

File: question-service/src/main/java/com/information/center/questionservice/service/CrunchifyTimeDiff.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
+package com.information.center.questionservice.service;
+
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+
+@Service
+public class CrunchifyTimeDiff {
```
Comment (kolyaattila) [inline question-service/src/main/java/com/information/center/questionservice/service/CrunchifyTimeDiff.java:8]: create tests for this class
